src/firefly/java/edu/caltech/ipac/astro/FITSTableReader.java

```diff
@@ -212,14 +213,28 @@ else if (strategy == FULLY_FLATTEN) {
      * @return List of star tables.
      * @throws IOException
      */
-    public static List getStarTableList(String fits_filename) throws IOException {
-
-        TableSequence tseq;
-        StarTableFactory stFactory = new StarTableFactory();
-        tseq = stFactory.makeStarTables(fits_filename, null);
-        List tList = new ArrayList();
-        for (StarTable tbl; (tbl = tseq.nextTable()) != null; ) {
-            tList.add(tbl);
+    public static List getStarTableList(String fits_filename) {
+
+        ArrayList<StarTable> tList = new ArrayList<>();
+
+        // disable long string for HeaderCard create while collecting tables from StarTableFactory to work around the exception error
+        // sent from nom.tam.fits.
+        FitsFactory.setLongStringsEnabled(false);
+
+        try {
+            TableSequence tseq;
+            StarTableFactory stFactory = new StarTableFactory();
+            tseq = stFactory.makeStarTables(fits_filename, null);
+
+            for (StarTable tbl; (tbl = tseq.nextTable()) != null; ) {
+                DescribedValue dValue = tbl.getParameterByName("ZIMAGE");   // check if 'BINTABLE' is actually compressed image HDU
```

Comment:
StarTable incorrectly pick up compressed image thinking it's a table?

Reply:
yes, they have XTENSION = BINTABLE in the header for tile compression HDU.